### PR TITLE
Release bundle builds should have --dev false flag

### LINKED ISSUE #4276: The release bundle builds should have --dev false flag
see:

https://github.com/microsoft/react-native-windows/blob/4a1c8bb418ec91302880b2cea89a236eaed4dbc2/packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj#L179

https://github.com/microsoft/react-native-windows/blob/744a02c3997197082e35efa74b55775fc8a7f25b/vnext/local-cli/generator-windows/templates/cpp/proj/MyApp.vcxproj#L163

If running this command:
`npx react-native bundle --help`

you can see its defaulting to true
...
 --dev [boolean]                    If false, warnings are disabled and the bundle is minified (default: true)
...

This is what can help lead to some 'yellow box' warnings appearing in release builds.

### PR DESCRIPTION
**Pull request by acoates-ms**
Fixes #4276

The default template creates a bundle file when doing a release build ready for the store.  But that bundle is a debug bundle.  This updates our bundle options in Bundle.props to align with the customization available on android in the default react-native gradle build.
https://github.com/facebook/react-native/blob/894f6b3e744679769ba0f6b83bae89354f0f1914/react.gradle#L170

Also updated the project file of the template to correctly handle react-native-windows being hoisted

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4603)

### CODE REVIEW

File: packages/playground/windows/playground/Playground.vcxproj [hunk 1]
```diff
@@ -172,8 +172,10 @@
 
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <PropertyGroup>
+    <UseDevBundle Condition="'$(Configuration)' == 'Debug'">true</UseDevBundle>
+    <UseDevBundle Condition="'$(Configuration)' != 'Debug'">false</UseDevBundle>
     <BundleCommand>
```
Comment (asklar) [inline packages/playground/windows/playground/Playground.vcxproj:177]: this shouldn't really be up to the app, it should just be a property you inherit from Import-ing one of our props

File: vnext/PropertySheets/Bundle.Common.targets [hunk 2]
```diff
@@ -8,7 +8,7 @@
```
Comment (asklar) [inline vnext/PropertySheets/Bundle.Common.targets:8]: props are supposed to be Imported by the app project. The idea is in the app project you'd do

Import props

override/add your own properties

Import targets

Comment (acoates-ms) [reply vnext/PropertySheets/Bundle.Common.targets]: @asklar 
I moved some things around.  But I'm not 100% sure this is correct.  Should the BundleCommand be defined in the props file at all?  Or should it just be moved into the targets file?  The way it currently is, if you're overriding something like BundleEntryFile you have to do it before you import the props, which is not how you described it above.

File: packages/E2ETest/windows/TreeDumpLibrary/TreeDumpLibrary.csproj [hunk 3]
```diff
@@ -116,7 +116,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
-      <Version>6.2.8</Version>
+      <Version>6.2.10</Version>
```
Comment (asklar) [inline packages/E2ETest/windows/TreeDumpLibrary/TreeDumpLibrary.csproj:119]: unrelated?

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCS/SampleAppCS.csproj [hunk 4]
```diff
@@ -174,13 +174,9 @@
...
   </PropertyGroup>
-  <PropertyGroup>
-    <BundleCommand>
-      npx --no-install yarn run bundle-cs
-    </BundleCommand>
-  </PropertyGroup>
-  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.targets" />
   <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />
+  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.props" />
+  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.targets" />
```
Comment (asklar) [inline packages/microsoft-reactnative-sampleapps/windows/SampleAppCS/SampleAppCS.csproj:179]: here in this C# project we use Bundle.targets but in C++ we use Bundle.cpp.targets. Why the difference? Can we just use one targets file and if we need to have different behavior it can figure it out for us?

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj [hunk 5]
```diff
@@ -198,11 +198,7 @@
     <Import Project="..\packages\Microsoft.UI.Xaml.2.3.191129002\build\native\Microsoft.UI.Xaml.targets" Condition="Exists('..\packages\Microsoft.UI.Xaml.2.3.191129002\build\native\Microsoft.UI.Xaml.targets')" />
     <Import Project="..\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.targets" Condition="Exists('..\packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.targets')" />
   </ImportGroup>
-  <PropertyGroup>
-    <BundleCommand>
-      npx --no-install yarn run bundle-cpp
-    </BundleCommand>
-  </PropertyGroup>
+  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.props" />
```
Comment (asklar) [inline packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleAppCpp.vcxproj:201]: normally you'd put all the imported props files before any propertygroup. I think this is the same end result in this case as you don't seem to be overwriting/overriding anything in the bundle.props, but it'd be worth sticking to the layout

File: packages/playground/windows/playground/Playground.vcxproj [hunk 6]
```diff
@@ -171,13 +171,11 @@
   </ItemGroup>
 
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <Import Project="$(ReactNativeWindowsDir)\PropertySheets\Bundle.props" />
```
Comment (asklar) [inline packages/playground/windows/playground/Playground.vcxproj:174]: move up before targets (really before the first PropertyGroup)

File: vnext/PropertySheets/Bundle.props [hunk 7]
```diff
@@ -6,10 +6,49 @@
...
+      UseBundle causes the bundle to be created and included inside the application install.  And defines the BUNDLE #define in code,
+      which causes the app to load from the bundle file rather than loading the bundle from a bundle server
+    -->
     <UseBundle Condition=" '$(Configuration)' == 'Debug' and '$(UseBundle)' == '' ">false</UseBundle>
     <UseBundle Condition=" '$(Configuration)' == 'Release' and '$(UseBundle)' == '' ">true</UseBundle>
-    <BundleContentRoot Condition=" '$(BundleContentRoot)' == '' ">Bundle</BundleContentRoot>
-    <BundleContent Condition = " '$(BundleContent)' == '' and '$(BundleContentRoot)' != '' ">$(BundleContentRoot)\**\*</BundleContent>
+
+    <!-- Root directory where bundle assets will be copied to, be sure to update BundleContent if you change this -->
+    <BundleContentRoot Condition=" '$(BundleContentRoot)' == '' ">$([MSBuild]::NormalizePath('$(ProjectDir)\Bundle'))</BundleContentRoot>
```
Comment (asklar) [inline vnext/PropertySheets/Bundle.props:18]: I *think* ProjectDir is a VS thing whereas MSBuildProjectDirectory is the MSBuild property - are you seeing this work on the command line too?
